test/integration/controller_test.go

```diff
@@ -435,26 +495,29 @@ func TestCAPIClusterCreate(t *testing.T) {
 					return
 				}
 				machineSets[i] = ms
+
 				for _, role := range ms.Spec.Template.Spec.Roles {
 					if role == capicommon.MasterRole {
-						if masterMachinSet != nil {
-							t.Fatalf("multiple master machinesets: %s and %s", masterMachinSet.Name, ms.Name)
+						if masterMachineSet != nil {
+							t.Fatalf("multiple master machinesets: %s and %s", masterMachineSet.Name, ms.Name)
 						}
-						masterMachinSet = ms
+						masterMachineSet = ms
 					}
 				}
-				if masterMachinSet == nil {
-					t.Fatalf("no master machineset")
-				}
+				coClusterSpec.MachineSets = append(coClusterSpec.MachineSets, clustopClusterMS)
```

Comment:
The machinesets have already been added to `coClusterSpec.MachineSets`. This will add them again.